blockchain/blockchain.go

```diff
@@ -159,8 +159,7 @@ type BlockChain struct {
 	scope         event.SubscriptionScope
 	genesisBlock  *types.Block
 
-	mu      sync.RWMutex // global mutex for locking chain operations
-	chainmu sync.RWMutex // blockchain insertion lock
+	mu sync.RWMutex // global mutex for locking chain operations
```

Comment:
@hqjang-pepper In Ethereum, `mu` is removed, but this PR removes `chainmu`. Do you have any reason for this?
Furthermore, in the given Ethereum PR, with removing `mu`, some logic is modified such as `WriteBlockWithState`. Is this okay?

Reply:
Fixed some part of `WriteBlockWithState`. Thanks for your advice.
`mu` of Klaytn has exactly same role with `chainmu` in Ethereum. Just naming issue. 
Since mutex in blockchain.go is protecting globally-used fields , not just chain insertion. 
So I left `mu` instead of `chainmu`.

Reply:
If it is not an important reason, please consider using the same name with Ethreeum for compatibility the next time. It would be better to cherry-pick their improvements.